test/integration/scheduler/framework_test.go

```diff
@@ -1885,10 +1884,11 @@ func TestPreemptWithPermitPlugin(t *testing.T) {
 	// }
 
 	if err := wait.Poll(200*time.Millisecond, wait.ForeverTestTimeout, func() (bool, error) {
-		_, err := getPod(testCtx.ClientSet, waitingPod.Name, waitingPod.Namespace)
-		return apierrors.IsNotFound(err), nil
+		w := false
+		permitPlugin.fh.IterateOverWaitingPods(func(wp framework.WaitingPod) { w = true })
+		return !w, nil
 	}); err != nil {
-		t.Error("Expected the waiting pod to get preempted and deleted")
+		t.Error("Expected the waiting pod to get preempted")
 	}
 
```

Comment:
Can you add a step to call API server to verify the waited Pod is **not** deleted physically?

Reply:
Bump.

Reply:
is there another test for non-permit case were we do verify that the pod is deleted?

Reply:
we can add another low-priority regular Pod and let it be running, so that the preemptor needs to preempt two pods, then we can verify one is removed from waitingMap, and the other is deleted physically.

Reply:
test added